packages/cli/lib/artifact-generator.js

```diff
@@ -99,18 +99,42 @@ module.exports = class ArtifactGenerator extends BaseGenerator {
        * Both those properties must be present for this to happen. Optionally,
        * this can be disabled even if the properties are present by setting:
        * this.artifactInfo.disableIndexUpdate = true;
+       *
+       * Multiple indexes / files can be updated by providing an array of
+       * index update objects as follows:
+       * this.artifactInfo.indexesToBeUpdated = [{
+       *   dir: 'directory in which to update index.ts',
+       *   file: 'file to add to index.ts',
+       * }, {dir: '...', file: '...'}]
        */
-      if (
-        this.artifactInfo.outDir &&
-        this.artifactInfo.outFile &&
-        !this.artifactInfo.disableIndexUpdate
-      ) {
-        await updateIndex(this.artifactInfo.outDir, this.artifactInfo.outFile);
-        // Output for users
-        this.log(
-          chalk.green('   update'),
-          `${this.artifactInfo.relPath}/index.ts`,
-        );
+      if (!this.artifactInfo.disableIndexUpdate) {
```

Comment:
Could you please use reverse-logic and early return to reduce the amount of indentation and make the code easier to reason about?

```ts
if (this.artifactInfo.disableIndexUpdate) return;
```

I think we should move the code handling index updates into a new method, the `end` method is already too long.

```ts
class ArtifactGenerator extends BaseGenerator {
  // ...

  async end() {
    // ...

    if (generationStatus) {
      this._updateIndexFiles();
    }

    // User Output
    this.log();
    // ...
  }

  async _updateIndexFiles() {
    if (this.artifactInfo.disableIndexUpdate) return;
    // ...
  }
}
```

Thoughts?